haystack/nodes/prompt/invocation_layer/amazon_bedrock.py

```
@@ -0,0 +1,308 @@
+            )
+        return str(resize_info["resized_prompt"])
+
+    @classmethod
+    def supports(cls, model_name_or_path, **kwargs):
+        supported_model_ids = cls.SUPPORTED_MODELS.keys()
+        model_supported = model_name_or_path in supported_model_ids
+        if not model_supported or not cls.aws_configured(**kwargs):
+            return False
+
+        try:
+            session = cls.get_aws_session(**kwargs)
+            bedrock = session.client("bedrock")
+            foundation_models_response = bedrock.list_foundation_models(byOutputModality="TEXT")
+            available_model_ids = [entry["modelId"] for entry in foundation_models_response.get("modelSummaries", [])]
+        except AWSConfigurationError as e:
+            raise AmazonBedrockConfigurationError(message=e.message) from e
+        except Exception as e:
+            raise AmazonBedrockConfigurationError(
+                "Could not connect to Amazon Bedrock. Make sure the AWS environment is configured correctly."
+            ) from e
+
+        model_available = model_name_or_path in available_model_ids
+        if not model_available:
+            raise AmazonBedrockConfigurationError(
+                f"The model {model_name_or_path} is not available in Amazon Bedrock. "
+                f"Please make sure the model is available in the configured AWS region and you've been granted access."
+            )
+
+        return model_supported
+
+    def supports_streaming(self, model_name_or_path):
+        foundation_models_response = self.bedrock.list_foundation_models(byOutputModality="TEXT")
+        available_model_ids = [
+            entry["modelId"]
+            for entry in foundation_models_response.get("modelSummaries", [])
+            if entry.pop("responseStreamingSupported", None)
+        ]
+        if model_name_or_path not in available_model_ids:
+            raise AmazonBedrockConfigurationError(f"{model_name_or_path} does not offer streaming support")
```

Please make this method returning a `bool`: we can throw the exception in `supports`. Also please add type annotations to params. Use `get` instead of `pop` please.

---

You can pass the bedrock client from `supports`.